homeassistant/components/elevenlabs/config_flow.py

```diff
@@ -0,0 +1,205 @@
+                    CONF_VOICE, description={"suggested_value": first_voice_id}
+                ): SelectSelector(
+                    SelectSelectorConfig(
+                        options=[
+                            SelectOptionDict(label=voice_name, value=voice_id)
+                            for voice_id, voice_name in self.voices.items()
+                        ]
+                    )
+                ),
+                vol.Required(
+                    CONF_MODEL, description={"suggested_value": DEFAULT_MODEL}
+                ): SelectSelector(
+                    SelectSelectorConfig(
+                        options=[
+                            SelectOptionDict(label=model_name, value=model_id)
+                            for model_id, model_name in self.models.items()
+                        ]
+                    )
+                ),
+            },
+        )
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Handle the initial step."""
+        errors = {}
+        if user_input is None:
+            return self.async_show_form(
+                step_id="user", data_schema=STEP_USER_DATA_SCHEMA_NO_AUTH
+            )
+        # Validate auth, get voices
+        try:
+            self.voices, self.models = await get_voices_models(user_input[CONF_API_KEY])
+        except ApiError:
+            errors[CONF_API_KEY] = "ElevenLabs API responded with an Error!"
+        if errors:
+            return self.async_show_form(
+                step_id="user", data_schema=STEP_USER_DATA_SCHEMA_NO_AUTH, errors=errors
+            )
+
+        self.user_info = user_input
+        return await self.async_step_voice()
+
+    async def async_step_voice(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Handle the voice selection step."""
+        if user_input is None:
+            return self.async_show_form(
+                step_id="voice", data_schema=self._get_user_schema_authenticated()
+            )
+        # Add api_key to user input
+        assert self.user_info is not None
+        user_input[CONF_API_KEY] = self.user_info[CONF_API_KEY]
+
+        return self.async_create_entry(
+            title=self.models[user_input[CONF_MODEL]], data=user_input
+        )
+
+    @staticmethod
+    def async_get_options_flow(
+        config_entry: ConfigEntry,
+    ) -> OptionsFlow:
+        """Create the options flow."""
+        return ElevenLabsOptionsFlow(config_entry)
+
+
+class ElevenLabsOptionsFlow(OptionsFlow):
+    """ElevenLabs options flow."""
+
+    # id -> name
+    voices: dict[str, str] = {}
+    models: dict[str, str] = {}
+
+    def __init__(self, config_entry: ConfigEntry) -> None:
+        """Initialize options flow."""
+        self.config_entry = config_entry
+        self.api_key: str = self.config_entry.data[CONF_API_KEY]
+        self.voices_models_loaded = False
```

Comment:
we don't need this. We can detect that if `self.voices` is None, it's not loaded.

Reply:
Good point. Fixed.